### ISSUE TITLE
Reorganize Che repository to simplify location of agent storage

### ISSUE THREAD

**Issue by TylerJewell** [eclipse-che/che#3437]
We have started to add in various agents into Che. The code and packaging for these agents are located in different locations in the repository today.  We should centralize all of the agent code into a single location.

1. We would have a `/agents` folder in the root of the repository.
2. `exec-agent` should be moved to `/agents/exec`.
3. We should investigate and explore whether ws-agent, ssh agent, sync agent, and LSP agents can also be moved to a subfolder of `/agents`.

**Comment by skabashnyuk** [comment]
Our proposal is something like this 
```
/agents
/agents/ws
/agents/exec
/agents/ssh
/plugins/plugin-1/server
/plugins/plugin-1/client
/plugins/plugin-1/shared
/plugins/plugin-1/agent
```
LSP is looks more like part of some language plugin

**Comment by TylerJewell** [comment]
Can you talk a bit more on lsp.  When I look at the files it seems that it is both a plugin and an agent. And ultimately each ls is packaged using shared plugin but dedicated zip for the agent itself.  So shouldn't we have both?

**Comment by TylerJewell** [comment]
Also, @skabashnyuk - as part of this issue, I think we should move `wsmaster` to `server`.  I think that will be clearer for people.

Also, after thinking about it, agree with your proposal on plugins. You are correct in that there are agents that are part of a plugin and agents that are just more cross-cutting.

I would like this to be in the next sprint - do you have an issue with that?

**Comment by skabashnyuk** [comment]
We would like to stay with ```wsmaster``` name and narrow the scope of this task to ```simplify location of agent storage```

**Comment by TylerJewell** [comment]
Ok for this issue.  But at least create another issue for the renaming of wsmaster to server as I think that is simpler name.

**Comment by skabashnyuk** [comment]
Continue refactoring we be done in https://github.com/eclipse/che/issues/3720

**Comment by TylerJewell** [comment]
@skabashnyuk - should we reopen this?  I ask because the folder that was created did not have the right names.  It was `exec-agent` instead of `/agents/exec`, for example.

**Comment by skabashnyuk** [comment]
We have pr for rename https://github.com/eclipse/che/pull/3775 it will be merged in 5.2
